### PR TITLE
Add @mapbox/mapbox-gl-supported libdef

### PR DESCRIPTION
**Pull request by NameFILIP**
- Link to GitHub or NPM: https://github.com/mapbox/mapbox-gl-supported
- Type of contribution: new definition

### CODE REVIEW

**Comment by Brianzchen** [comment]
Hey there, thanks for submitting. Your definition struct is a bit malformed which is why your tests are failing. Maybe try using `./create_def.sh` as referenced [here](https://github.com/flow-typed/flow-typed/blob/master/CONTRIBUTING.md#making-a-contribution)

File: definitions/npm/@mapbox/mapbox-gl-supported_v2.x.x/flow_v0.25.x-/mapbox-gl-supported_v2.x.x.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// @flow
```
Comment (Brianzchen) [inline definitions/npm/@mapbox/mapbox-gl-supported_v2.x.x/flow_v0.25.x-/mapbox-gl-supported_v2.x.x.js:1]: This line isn't necessary as it gets pulled into definition dir later

File: definitions/npm/@mapbox/mapbox-gl-supported_v2.x.x/flow_v0.25.x-/mapbox-gl-supported_v2.x.x.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+declare module '@mapbox/mapbox-gl-supported' {
+  declare type OptionsT = {
+    failIfMajorPerformanceCaveat: boolean,
+    ...
+  };
+  declare export function supported(options?: OptionsT): boolean;
+  declare export function notSupportedReason(options?: OptionsT): ?string;
```
Comment (Brianzchen) [inline definitions/npm/@mapbox/mapbox-gl-supported_v2.x.x/flow_v0.25.x-/mapbox-gl-supported_v2.x.x.js:7]: Based on example from npm that recommends usage as

```
var mapboxglSupported = require('@mapbox/mapbox-gl-supported');

if (mapboxglSupported.supported()) {
    ...
}
```

Perhaps using `module.exports` will give more compatibility. This style should work without any changes to your tests which I assume is how you're consuming in your app.

```suggestion
  module.exports = {|
    supported: (options?: OptionsT) => boolean,
    notSupportedReason: (options?: OptionsT) => ?string,
  |};
```

**Review by Brianzchen — APPROVED** [review]
Looking great, just a bit of feedback, let me know what you think

File: definitions/npm/@mapbox/mapbox-gl-supported_v2.x.x/flow_v0.25.x-/mapbox-gl-supported_v2.x.x.js (hunk 2 above)
Comment (NameFILIP) [reply definitions/npm/@mapbox/mapbox-gl-supported_v2.x.x/flow_v0.25.x-/mapbox-gl-supported_v2.x.x.js]: Yep, I think that makes sense. Updated.

**Comment by Brianzchen** [comment]
😅 correctly fails now with that extra prop

**Comment by NameFILIP** [comment]
@Brianzchen yep, moved it to the failing `it`s

File: definitions/npm/@mapbox/mapbox-gl-supported_v2.x.x/flow_v0.25.x-/mapbox-gl-supported_v2.x.x.js [hunk 3]
```diff
@@ -0,0 +1,8 @@
+declare module '@mapbox/mapbox-gl-supported' {
+  declare type OptionsT = {
+    failIfMajorPerformanceCaveat: boolean,
+    ...
+  };
```
[the remark this answers is not in the document]
Comment (NameFILIP) [reply definitions/npm/@mapbox/mapbox-gl-supported_v2.x.x/flow_v0.25.x-/mapbox-gl-supported_v2.x.x.js]: Done

**Comment by NameFILIP** [comment]
@Brianzchen, thank you for the feedback and for quickly accepting the contribution!

**Pull request merged**